src/rust/metric-forwarder/src/cloudwatch_send.rs

```diff
@@ -0,0 +1,177 @@
+use crate::cloudwatch_logs_parse::Stat;
+use crate::error::MetricForwarderError;
+use async_trait::async_trait;
+use futures::future;
+use log::warn;
+use rayon::prelude::*;
+use rusoto_cloudwatch::PutMetricDataError;
+use rusoto_cloudwatch::{CloudWatch, MetricDatum, PutMetricDataInput};
+use rusoto_core::RusotoError;
+use statsd_parser;
+use statsd_parser::Metric;
+
+mod units {
+    // strings accepted by CloudWatch MetricDatum.unit
+    pub const COUNT: &'static str = "Count";
+    pub const MILLIS: &'static str = "Milliseconds";
+}
+
+type PutResult = Result<(), RusotoError<PutMetricDataError>>;
+
+#[async_trait]
+pub trait CloudWatchPutter {
+    // a subset of trait CloudWatch with the 1 function we want
+    async fn put_metric_data(&self, input: PutMetricDataInput) -> PutResult;
+}
+
+#[async_trait]
+impl<T> CloudWatchPutter for T
+where
+    T: CloudWatch + Sync + Send,
+{
+    async fn put_metric_data(&self, input: PutMetricDataInput) -> PutResult {
+        self.put_metric_data(input).await
```

Comment:
I'm a little worried about infinite recursion here. It might be better to do this via explicit universal call syntax

```rust
CloudWatch::put_metric_data(&self, input).await
```